### PR TITLE
feat: Network api

### PR DESCRIPTION
**Pull request by Gurgel100**
This PR adds a binding to the wasm network api introduced with SU11.

### CODE REVIEW

File: msfs/src/network.rs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        mut self,
+        post_field: Option<CString>,
+        request: unsafe extern "C" fn(
+            *const ::std::os::raw::c_char,
+            *mut sys::FsNetworkHttpRequestParam,
+            sys::HttpRequestCallback,
+            *mut ::std::os::raw::c_void,
+        ) -> sys::FsNetworkRequestId,
+    ) -> Option<NetworkRequest> {
+        let raw_post_field = post_field.map_or(ptr::null_mut(), |f| f.into_raw());
```
[the remark this answers is not in the document]
Comment (Gurgel100) [reply msfs/src/network.rs:75]: 1. I think you mean `f.as_c_str().as_ptr()`?
2. Why would this allow to remove the drop later?

[the remark this answers is not in the document]
Comment (devsnek) [reply msfs/src/network.rs:74]: 1. yes
2. because it doesn't consume the `CString` like `into_raw` does.

[the remark this answers is not in the document]
Comment (Gurgel100) [reply msfs/src/network.rs:74]: Thanks for explaining. Done.

File: msfs/src/network.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    ) {
+        if !user_data.is_null() {
+            let callback: Box<NetworkCallback> = unsafe { Box::from_raw(user_data as *mut _) };
+            callback(NetworkRequest(request_id), status_code);
+        }
+    }
+}
+
+/// The states in which a network request can be in
+#[derive(Clone, Copy)]
```
Comment (pepperoni505) [inline msfs/src/network.rs:126]: Maybe add `PartialEq, Eq` to this? Would be useful for checking the request state on the callback

Comment (Gurgel100) [reply msfs/src/network.rs:126]: Done.

File: msfs/Cargo.toml [hunk 3]
```diff
@@ -9,6 +9,7 @@ license = "MIT"
 [dependencies]
 msfs_derive = { path = "../msfs_derive", version = "0.2.0" }
 futures = "0.3"
+libc = "0.2"
```
[the remark this answers is not in the document]
Comment (Gurgel100) [reply msfs/Cargo.toml:12]: It's used at lines 177 & 202 in network.rs. Imports are not needed anymore in "new" Rust.

[the remark this answers is not in the document]
Comment (pepperoni505) [reply msfs/Cargo.toml:12]: Ah ok, thanks for clarifying that

**Pull request merged**